tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/bindgen/BindgenTestCase.java

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * Integration tests for the bindgen tool.
+ *
+ * @since 1.2.5
+ */
+public class BindgenTestCase extends BaseTest {
+
+    private Path tempProjectsDirectory;
+    private Path projectPath;
+    private BMainInstance balClient;
+
+    @BeforeClass()
+    public void setUp() throws IOException, BallerinaTestException {
+        tempProjectsDirectory = Files.createTempDirectory("bal-test-integration-bindgen-");
+        // copy TestProject to a temp
+        Path originalTestProj1 = Paths.get("src", "test", "resources", "bindgen", "TestProject")
+                .toAbsolutePath();
+        projectPath = tempProjectsDirectory.resolve("module1");
+        copyFolder(originalTestProj1, projectPath);
+        balClient = new BMainInstance(balServer);
+    }
+
+    /**
+     * Generate Ballerina bindings covering multiple scenarios and build the project.
+     *
+     * @throws BallerinaTestException Error when executing the commands.
+     */
+    @Test(description = "Test the bindgen command functionality.")
+    public void bindgenTest() throws BallerinaTestException {
+        String buildMsg = "target/bin/module1.jar";
+        LogLeecher buildLeecher = new LogLeecher(buildMsg);
+
+        String bindgenMsg = "class could not be generated.";
+        LogLeecher bindgenLeecher = new LogLeecher(bindgenMsg, ERROR);
+
+        String runMsg1 = "Array util tests successful.";
+        String runMsg2 = "Exception handling tests successful.";
+        LogLeecher runLeecher1 = new LogLeecher(runMsg1);
+        LogLeecher runLeecher2 = new LogLeecher(runMsg2);
+
+        String[] args = {"-mvn=org.yaml:snakeyaml:1.25", "-o=./src/module1", "org.yaml.snakeyaml.Yaml",
+                "java.lang.Object", "java.lang.String", "java.lang.Class", "java.lang.StringBuffer",
+                "java.lang.ArithmeticException", "java.lang.AssertionError", "java.lang.StackTraceElement",
+                "java.lang.Character$Subset", "java.lang.Character", "java.io.File", "java.lang.Short",
+                "java.io.FileInputStream", "java.io.OutputStream", "java.io.FileOutputStream", "java.util.HashSet",
+                "java.util.Set", "java.util.Iterator"};
+        try {
+            balClient.runMain("bindgen", args, null, new String[]{},
+                    new LogLeecher[]{bindgenLeecher}, projectPath.toString());
+            throw new BallerinaTestException("Contains classes not generated.");
+        } catch (BallerinaTestException e) {
+            if (bindgenLeecher.isTextFound()) {
+                throw new BallerinaTestException("Contains classes not generated.", e);
+            }
+        }
+        balClient.runMain("build", new String[]{"-a"}, null, new String[]{},
+                new LogLeecher[]{buildLeecher}, projectPath.toString());
+        balClient.runMain("run", new String[]{"target/bin/module1.jar"}, null, new String[]{},
+                new LogLeecher[]{runLeecher1, runLeecher2}, projectPath.toString());
```

Comment:
Don't we need to validate the output of the run command after this? i.e. `runLeeecher1.waitForText(5000)` and `runLeeecher2.waitForText(5000)`

Reply:
Yes, will change this.